storage-integration/src/main/java/com/dsi/storage/client/minio/MinioStorageService.java

```diff
@@ -23,88 +22,92 @@
  * Example: "my-bucket/folder1/folder2/6cbd360f-df93-48eb-901b-87e97a5ddb8e"
  */
 public class MinioStorageService implements StorageClient {
-    private static final String endpoint = System.getenv("STORAGE_ENDPOINT");
-    private static final String accessKey = System.getenv("STORAGE_ACCESS_KEY");
-    private static final String secretKey = System.getenv("STORAGE_SECRET_KEY");
-    private static final long partSize = (System.getenv("STORAGE_PART_SIZE") != null)
-            ? Long.parseLong(System.getenv("STORAGE_PART_SIZE"))
-            : 10485760L; // 10 MB default size
-
+    private final long partSize;
     private final MinioClient minioClient;
     private static final Logger logger = LoggerFactory.getLogger(MinioStorageService.class);
 
-    /**
-     * Constructs a MinioStorageService instance using environment variables for MinIO configuration.
-     */
-    public MinioStorageService() {
-        ValidationUtils.validateNotEmpty(endpoint, accessKey, secretKey);
+    public MinioStorageService(String endpoint, String accessKey, String secretKey, long partSize) {
         this.minioClient = MinioClient.builder()
                 .endpoint(endpoint)
                 .credentials(accessKey, secretKey)
                 .build();
+        this.partSize = partSize;
     }
 
     @Override
     public String upload(String fullPath, InputStream data, String contentType) throws StorageException {
         try {
-            String baseBucket = FilePathUtil.extractBaseBucket(fullPath);
-            String directoryBucketString = FilePathUtil.convertPathToDirectoryBucketString(fullPath);
+            String[] parts = PathUtil.splitPathForUpload(fullPath);
+            String baseBucket = parts[0];
```

Comment:
add empty array check in a separate method with PathUtil.splitPathForUpload(fullPath);